src/fraction.ts

```diff
@@ -64,10 +64,17 @@ export class Fraction {
   }
 
   toNumber() {
-    return (
-      parseInt(this.numerator.toString()) /
-      parseInt(this.denominator.toString())
-    );
+    let numerator = this.numerator.clone();
+    let denominator = this.denominator.clone();
+    // Prevent overflow by removing least significant bits
+    while (
+      parseInt(numerator.toString()) === Infinity ||
+      parseInt(denominator.toString()) === Infinity
+    ) {
+      numerator = numerator.ishrn(1);
+      denominator = denominator.ishrn(1);
```

Comment:
What happens when you have a fraction of `very_very_very_large / 1`, would this cause a divide by 0?

Reply:
I case we divide by 0 we would get Infinity as a result which I guess makes sense since we are overflowing the largest number?